### ISSUE TITLE
The non-intrusive toast isn't persisted in the Action Center on Windows 10

### ISSUE THREAD

**Issue by Tragetaschen** [git-for-windows/git#1455]
- [x ] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

I was updating from v2.15.1(2) to v2.16.1, so the toast came from 2.15

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

Windows 10 64-bit

 - What options did you set as part of the installation? Or did you choose the
   defaults?

I'm running with the Task Scheduler based update notification active.

### Details

The Task Scheduler currently fires of a Toast once every day (9:58 CET for me) and shows the Toast for 15 seconds. If you miss it, there's no clear path how to get the download: Either open the web page and download the installer manually, or wait for another day and hope for the best, or run the Task Scheduler task manually from Computer Management and then click "Yes" in the toast.

It would be great, if the toast could be found in the Action Center instead after it timed out similar to new mail notifications.

**Comment by dscho** [comment]
This is where the toast is created: https://github.com/git-for-windows/build-extra/blob/285cfd53e625e19279f13323dd483f4819900164/git-extra/git-update#L205

Maybe you can spot the problem?

**Comment by dscho** [comment]
> Maybe you can spot the problem?

I can't, and without help, I am afraid that we'll not get anywhere close to the resolution.

**Comment by shiftkey** [comment]
Looking at the [initial description](https://blogs.msdn.microsoft.com/tiles_and_toasts/2015/07/08/toast-notification-and-action-center-overview-for-windows-10/) of how notifications and the Action Center work (this is from 2015, so it's probably not up-to-date), these comment stands out:

> **A toast notification, after being swiped away or ignored by the user, automatically goes into action center**. Action center is a system area that allows users to see previously missed or ignored toast notifications, and to quickly access system settings such as Wi-Fi and Bluetooth.
> ...
> A toast notification will stay in action center for 7 days until it expires (which will then be removed from action center), **unless an earlier expiration time is specified during the creation of the toast**. See `ExpirationTime` for more information.

My psychic debugger thinks that setting the expiration time to `15000ms` (aka 15mins) is causing it to disappear from the action center.

**Comment by DotNetSparky** [comment]
LOL, @shiftkey beat me to it...  I'll add, it sounds like there IS a way to do it, but requires a little more work.  I found various sources pointing to here for making a persistent notification in the action center...

https://blogs.msdn.microsoft.com/tiles_and_toasts/2015/10/16/quickstart-handling-toast-activations-from-win32-apps-in-windows-10/

**Comment by dscho** [comment]
> I'll add, it sounds like there IS a way to do it, but requires a little more work.

Yep, and I thought that we settled on waiting in the `git update-git-for-windows` script (formerly known as `git update`) for feedback so that we did not have to implement a full-blown COM server (in which case I would not even know how to hand through the information to `wintoast.exe` what we want to do if the user hits "Update").

It is quite sad, but a simple reality, that I won't have more time to spend on the updater feature after sinking several dozens of hours into that project. Not unless I have help, that is...

**Comment by DotNetSparky** [comment]
> It is quite sad, but a simple reality, that I won't have more time to spend on the updater feature after sinking several dozens of hours into that project. Not unless I have help, that is...

I understand and can totally appreciate that.  I might mess with it a little myself, if nothing else it'd be a learning experience (I haven't messed with Windows' toast notifications before).  No promises, it *is* COM we're talking about after all... :-)

**Comment by dscho** [comment]
Thanks!
